examples/multilang/mycomponent/index.js

```diff
@@ -0,0 +1,35 @@
+const pulumi = require("@pulumi/pulumi");
+const aws = require("@pulumi/aws");
+
+// TODO: This should be done in the AWS library.  Also, not clear that providers work correctly
+// currently - we appear to end up with class to `Check` a provider before it has been configured -
+// I suspect that the proxy is getting confused as a separate identity even though it's the same
+// URN.
+pulumi.runtime.registerProxyConstructor("pulumi:providers:aws", aws.Provider)
```

Comment:
why no lambda passed to this?

Reply:
`aws.Provider` is a resource constructor already.  In general, this should be the "normal" pattern - that you just provide the constructor function directly.  Only if the constructor function is somehow "non-standard" would you need to wrap in a lambda.